webapp/utils/aws/ecs.py

```diff
@@ -87,43 +89,95 @@ def monitor_task(self, task: str, timeout: int = 600) -> dict:
         while True:
             if time.time() - start > timeout:
                 raise ECSTaskRuntimeExceededTimeoutError(timeout)
-            response = self.client.describe_tasks(cluster=self.cluster, tasks=[task])
-            task_run_details = self.parse_run_details(response)  # type: ignore[arg-type]
-            task_status = task_run_details["status"]
-
-            message = f"Status for task {task}: {task_status}"
-            logger.info(message)
-
+            task_status = self.get_task_status(task_id)
             if task_status == "STOPPED":
                 logger.info("Task run has completed.")
                 break
             time.sleep(5)
-        return task_run_details
 
-    def task_exists(self) -> bool | None:
-        response = self.client.list_task_definitions(
-            familyPrefix=self.task_family, sort="DESC"
-        )
-        valid_tasks = [task.split("/")[-1] for task in response["taskDefinitionArns"]]
+    def get_task_status(self, task_id: str) -> str:
+        """Get status of an ECS task.
 
-        if self.task_family_revision in valid_tasks:
-            return True
-        raise ECSTaskDefinitionDoesNotExistError(self.task_definition)
+        For more information on the different stages of the ECS task lifecycle,
+        see https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-lifecycle-explanation.html.
 
-    def parse_run_details(self, details: dict) -> dict:
-        """Parse task run details.
+        Args:
+            task_id (str): ECS task ID.
 
-        Note: While the 'taskArn' and 'lastStatus' parameters are automatically populated
-              when an ECS task is kicked off, other parameters -- like 'createdAt'
-              and 'stoppedAt' are populated as the task is running.
+        Returns:
+            str: Status of an ECS task, representing a stage of the task lifecycle.
+        """
+        if self.task_exists(task_id):
+            response = self.client.describe_tasks(cluster=self.cluster, tasks=[task_id])
+            task_status = response["tasks"][0]["lastStatus"]
+            message = f"Status for task {task_id}: {task_status}"
+            logger.info(message)
+            return task_status
+        raise ECSTaskDoesNotExistError(task_id)
 
-        For more information on the response syntax provided by boto3.ECS.Client.run_task,
-        see https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecs/client/run_task.html.
+    def get_active_tasks(self) -> dict:
```

Comment:
Nice method!  Easy to parse and understand, and can see the utility in a couple of different situations.